src/Illuminate/Queue/RedisQueue.php

```diff
@@ -88,7 +88,16 @@ public function size($queue = null)
      */
     public function push($job, $data = '', $queue = null)
     {
-        return $this->pushRaw($this->createPayload($job, $this->getQueue($queue), $data), $queue);
+        if (is_null($this->blockFor)) {
+            return $this->pushRaw($this->createPayload($job, $this->getQueue($queue), $data), $queue);
+        }
+
+        $this->getConnection()->multi();
+        $this->getConnection()->rpush($this->getQueue($queue).':notify', '1');
+        $id = $this->pushRaw($this->createPayload($job, $this->getQueue($queue), $data), $queue);
+        $this->getConnection()->exec();
+
```

Comment:
`rpush queue:notify 1` is missing from  LuaScripts::migrateExpiredJobs().
Also I think this code is better to be moved to LuaScript as well.